docs/website/docs/general-usage/credentials/config_providers.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Configuration Providers
+description: Configuration dlt Providers
+keywords: [credentials, secrets.toml, secrets, config, configuration, environment
+      variables, provider]
+---
+
+# Configuration Providers
+## The provider hierarchy
+
+If function signature has arguments that may be injected, `dlt` looks for the argument values in
+providers. **The argument name is a key in the lookup**.
+
+Example:
+
+```python
+import dlt
+
+
+@dlt.source
+def google_sheets(
+    spreadsheet_id,
+    tab_names=dlt.config.value,
+    credentials=dlt.secrets.value,
+    only_strings=False
+):
+    sheets = build('sheets', 'v4', credentials=Services.from_json(credentials))
+    tabs = []
+    for tab_name in tab_names:
+        data = sheets.get(spreadsheet_id, tab_name).execute().values()
+        tabs.append(dlt.resource(data, name=tab_name))
+    return tabs
+```
+
+In case of `google_sheets()` it will look
+for: `tab_names`, `credentials` and `only_strings`.
+
+Each provider has its own key naming convention, and dlt is able to translate between them.
+
+Providers form a hierarchy. At the top are environment variables, then `secrets.toml` and
+`config.toml` files. Providers like Airflow/Google/AWS/Azure Vaults will be inserted **after** the environment
+provider but **before** `toml` providers.
+
+For example, if `spreadsheet_id` is found in environment variable `SPREADSHEET_ID`, `dlt` will not look in `toml` files
+and below.
+
+The values passed in the code **explicitly** are the **highest** in provider hierarchy. The **default values**
+of the arguments have the **lowest** priority in the provider hierarchy.
+
+> **Summary of the hierarchy:**
```

Comment:
">" creates a markdown blockquote. I believe it needs to be either removed or replaces an [admonition](https://docusaurus.io/docs/markdown-features/admonitions) if it's a sidenote.